File: Java/Anotherbrick.java
import java.util.Scanner;

class Anotherbrick {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int height = sc.nextInt();
        int width = sc.nextInt();
        int bricks = sc.nextInt();
        int totH = 0, totW = 0;

        while (bricks-- > 0 && totH <= height && totW <= width) {
            totW += sc.nextInt();
            if (totW == width) {
                totW = 0;
                totH++;
            }
            if (totH == height)
                break;
        }
        System.out.println(totH == height ? "YES" : "NO");
        sc.close();
    }
}

File: Kotlin/anotherbrick.kt
import java.util.Scanner

fun main() {
    val sc = Scanner(System.`in`)
    var (h, w, b) = readLine()!!.split(' ').map(String::toInt)
    var totH: Int = 0; var totW: Int = 0

    while (b-- > 0 && totH <= h && totW <= w) {
        totW += sc.nextInt()
        if (totW == w) {
            totW = 0
            totH++
        }
        if (totH == h) break;
    }

    if (totH == h) println("YES") else println("NO")
}